src/GafferImage/CreateViews.cpp

```diff
@@ -0,0 +1,196 @@
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//      * Redistributions of source code must retain the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer.
+//
+//      * Redistributions in binary form must reproduce the above
+//        copyright notice, this list of conditions and the following
+//        disclaimer in the documentation and/or other materials provided with
+//        the distribution.
+//
+//      * Neither the name of John Haddon nor the names of
+//        any other contributors to this software may be used to endorse or
+//        promote products derived from this software without specific prior
+//        written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "GafferImage/CreateViews.h"
+
+#include "GafferImage/BufferAlgo.h"
+#include "GafferImage/ImageAlgo.h"
+
+#include "Gaffer/ArrayPlug.h"
+#include "Gaffer/Context.h"
+#include "Gaffer/NameValuePlug.h"
+
+#include "boost/bind/bind.hpp"
+#include "boost/range/adaptor/reversed.hpp"
+
+using namespace std;
+using namespace Imath;
+using namespace IECore;
+using namespace Gaffer;
+using namespace GafferImage;
+
+//////////////////////////////////////////////////////////////////////////
+// CreateViews
+//////////////////////////////////////////////////////////////////////////
+
+GAFFER_NODE_DEFINE_TYPE( CreateViews );
+
+size_t CreateViews::g_firstPlugIndex = 0;
+
+CreateViews::CreateViews( const std::string &name )
+	:	ImageNode( name )
+{
+	storeIndexOfNextChild( g_firstPlugIndex );
+
+	NameValuePlugPtr element = new NameValuePlug( "left", new GafferImage::ImagePlug(), /* defaultEnabled = */ true, "view0" );
+	ArrayPlugPtr views = new ArrayPlug(
+		"views",
+		Plug::In,
+		element,
+		0,
+		std::numeric_limits<size_t>::max(),
+		Plug::Default,
+		/* resizeWhenInputsChange = */ false
+	);
+	addChild( views );
+
+	NameSwitchPtr s = new NameSwitch( "__switch" );
+	addChild( s );
+
+	s->setup( new ImagePlug() );
+	s->selectorPlug()->setValue( "${" + ImagePlug::viewNameContextName.string() + "}" );
```

Comment:
Hmm, that toggle would do the trick here.  The issue would be switches with integer indices, or even a named switched where the name was set by an expression instead of a string substitution.

Reply:
Still seems like a good idea, but now irrelevant to this PR, since for the new magic we don't tamper with the context vairable here.